src/NServiceBus.Core/Pipeline/Incoming/ReceiveFeature.cs

```diff
@@ -44,16 +44,14 @@ protected internal override void Setup(FeatureConfigurationContext context)
 
         class NoOpAdaper : ISynchronizedStorageAdapter
         {
-            public bool TryAdapt(OutboxTransaction transaction, out CompletableSynchronizedStorageSession session)
+            public Task<CompletableSynchronizedStorageSession> TryAdapt(OutboxTransaction transaction, ContextBag context)
             {
-                session = null;
-                return false;
+                return Task.FromResult<CompletableSynchronizedStorageSession>(null);
```

Comment:
I.e. Task.FromResult(default(CompletableSynchronizedStorageSession)) and assign it to a field